functions.sh

```diff
@@ -239,3 +239,54 @@ function sort_versions() {
 
   echo "${sorted[@]}"
 }
+
+function commit_range() {
+  local commit_id_end=${1}
+  shift
+  local commit_id_start=${1}
+
+  if [ -z "${commit_id_start}" ] && [ -z "${commit_id_end}" ]; then
+    echo "HEAD~1..HEAD"
+  elif [ -z "${commit_id_start}" ] && [[ "${commit_id_end}" =~ .. ]]; then
+    echo "${commit_id_end}"
+  elif [ -z "${commit_id_start}" ]; then
+    echo "${commit_id_end}~1..${commit_id_end}"
+  else
+    echo "${commit_id_end}..${commit_id_start}"
+  fi
+}
+
+function images_updated() {
+  local commit_range
+  local versions
+  local images_changed
+
+  commit_range="$(commit_range "$@")"
+
+  IFS=' ' read -ra versions <<<"$(
+    IFS=','
+    get_versions
+  )"
+  images_changed=$(git diff --name-only "${commit_range}" "${versions[@]}")
+
+  if [ -z "${images_changed}" ]; then
+    return 1
+  else
+    return 0
+  fi
+}
+
+function tests_updated() {
+  local commit_range
+  local test_changed
+
+  commit_range="$(commit_range "$@")"
+
+  test_changed=$(git diff --name-only "${commit_range}" test*)
+
+  if [ -z "${test_changed}" ]; then
+    return 1
+  else
+    return 0
+  fi
```

Comment:
Let's keep the origin version?
```sh
  if [ -z "${images_changed}" ]; then
    return 1
  fi
  return 0
```

Reply:
This is a new function, the original is above. Now we have one function to teat the images and another one to check the tests

Reply:
still the `else` is not needed

Reply:
OOOh right!

Reply:
Done!